src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java

```diff
@@ -12,39 +14,55 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Component
 public class VendorMapper {
-    private final LocationService locationService;
     private final LocationMapper locationMapper;
     private final ModelMapper modelMapper = new ModelMapper();
+    private final LocationRepository locationRepository;
 
     @Autowired
-    public VendorMapper(LocationService locationService, LocationMapper locationMapper) {
-        this.locationService = locationService;
+    public VendorMapper(LocationMapper locationMapper, LocationRepository locationRepository) {
         this.locationMapper = locationMapper;
+        this.locationRepository = locationRepository;
         configureModelMapper();
     }
 
     public VendorResponseVO toVO(Vendor vendor) {
-        var response = modelMapper.map(vendor, VendorResponseVO.class);
-        response.setLocation(locationMapper.toVO(vendor.getLocation()));
+        VendorResponseVO response = modelMapper.map(vendor, VendorResponseVO.class);
+
+        List<LocationResponseVO> locations = vendor.getLocations()
+                .stream().map(locationMapper::toVO)
+                .collect(Collectors.toList());
+
+        response.setLocations(locations);
         return response;
     }
 
     public Vendor toEntity(VendorRequestVO request) {
-        var vendor = modelMapper.map(request, Vendor.class);
+        Vendor vendor = modelMapper.map(request, Vendor.class);
         provideLocationDependencies(request, vendor);
         return vendor;
     }
 
-    public void update(Vendor vendor, VendorRequestVO request) {
-        provideLocationDependencies(request, vendor);
+    public void update(VendorRequestVO request, Vendor vendor) {
+        List<Location> locations = request.getLocationIds().stream()
+                .map(l -> locationRepository.findById(l).orElseThrow(() -> new EntityNotFoundException(Location.class, "id", l)))
+                .collect(Collectors.toList());
+
+        locations.removeIf(e -> vendor.getLocations().contains(e));
```

Comment:
What if we want to remove the existing linkage between the vendor and location during the vendor update?

Reply:
changed it